### PR TITLE
Add LLVM 14 support

### PR DESCRIPTION
**Pull request by MMory**

### CODE REVIEW

File: lib/PhasarLLVM/Utils/Annotation.cpp [hunk 1]
```diff
@@ -55,7 +56,8 @@ uint64_t VarAnnotation::getLine() const {
 llvm::StringRef VarAnnotation::retrieveString(unsigned Idx) const {
   if (const auto *ConstExpr = llvm::dyn_cast<llvm::ConstantExpr>(
           AnnotationCall->getArgOperand(Idx))) {
-    if (ConstExpr->isGEPWithNoNotionalOverIndexing()) {
+    if (true // ConstExpr->isGEPWithNoNotionalOverIndexing()
```
Comment (fabianbs96) [inline lib/PhasarLLVM/Utils/Annotation.cpp:59]:
```suggestion
    if (llvm::isa<llvm::GEPOperator>(ConstExpr)
```

File: lib/PhasarLLVM/Utils/Annotation.cpp [hunk 2]
```diff
@@ -109,7 +111,8 @@ llvm::StringRef GlobalAnnotation::retrieveString(unsigned Idx) const {
   const auto *AnnotationGepOp = AnnotationStruct->getOperand(Idx);
   if (const auto *ConstExpr =
           llvm::dyn_cast<llvm::ConstantExpr>(AnnotationGepOp)) {
-    if (ConstExpr->isGEPWithNoNotionalOverIndexing()) {
+    if (true // ConstExpr->isGEPWithNoNotionalOverIndexing()
```
Comment (fabianbs96) [inline lib/PhasarLLVM/Utils/Annotation.cpp:114]: Same as above

File: lib/PhasarLLVM/DataFlowSolver/Mono/Problems/InterMonoFullConstantPropagation.cpp [hunk 3]
```diff
@@ -79,13 +79,13 @@ InterMonoFullConstantPropagation::callFlow(
       llvm::isa<llvm::InvokeInst>(CallSite)) {
     const auto *CS = llvm::cast<llvm::CallBase>(CallSite);
     // early exit; varargs not handled yet
-    if (CS->getNumArgOperands() == 0 || Callee->isVarArg()) {
+    if (CS->arg_size() == 0 || Callee->isVarArg()) {
```
Comment (fabianbs96) [inline lib/PhasarLLVM/DataFlowSolver/Mono/Problems/InterMonoFullConstantPropagation.cpp:82]:
```suggestion
    if (CS->arg_empty() || Callee->isVarArg()) {
```

File: lib/PhasarLLVM/Utils/Annotation.cpp (hunk 1 above)
Comment (MMory) [reply lib/PhasarLLVM/Utils/Annotation.cpp:59]: Is that equivalent? What is the purpose of this if anyhow?

Comment (fabianbs96) [reply lib/PhasarLLVM/Utils/Annotation.cpp:59]: No, it is not equivalent, but as close as we can get. 
When you pass a constant string to a function, the string data is a global variable of an array type, so you always pass a constant GEP to that global instead. 

In the ctor of VarAnnotation we don't check that the AnnotationCall is indeed a call to an llvm.var.annotation intrinsic, so in theory this class could be reused for some other purpose. However, in our use-case, this if-check should always succeed

**Review by vulder — APPROVED** [review]
Except for the `isGEPWithNoNotionalOverIndexing` issue, this looks fine to me.

**Review by fabianbs96 — APPROVED** [review]

**Pull request merged**